src/integral.jl

```diff
@@ -68,57 +23,38 @@ contrast, increasing `FP` to e.g. `BigFloat` will typically increase precision
 """
 function integral end
 
-# If only f and geometry are specified, default to HAdaptiveCubature
+# If only f and geometry are specified, select default algorithm
 function integral(
     f::F,
     geometry::G
 ) where {F<:Function, G<:Meshes.Geometry}
-    _integral(f, geometry, HAdaptiveCubature())
-end
-
-# If algorithm is HAdaptiveCubature, use the generalized n-dim solution
-function integral(
-    f::F,
-    geometry::G,
-    settings::HAdaptiveCubature
-) where {F<:Function, G<:Meshes.Geometry}
-    _integral(f, geometry, settings)
+    N = Meshes.paramdim(geometry)
+    rule = (N == 1) ? GaussKronrod() : HAdaptiveCubature()
```

Comment:
One question about the naming: Now it seems like we have three different names for essentially the same thing: `settings`, `algorithm` (in the docstring) and now also `rule`. Is there some meaning behind the different names or could we always just use the same name to avoid confusion?

Reply:
We are on the exact same wavelength here. One of the changes I want to roll in with the other breaking `v0.14` changes in an upcoming PR is to get all of this nomenclature aligned. I’m thinking the abstract type should be `IntegrationRule` and “rule” should be used in place of all prior uses of algorithm, setting, etc.

Reply:
Ok, if this will be aligned in a future PR, I'm fine with that! Thanks for the clarification.